### PR TITLE
Context metadata adjusted for the transformed datetime sdtype

### LINKED ISSUE #1485: In `PARSynthesizer` I cannot pass in datetime context (`InvalidDataError` during fitting)
_I'm filing this issue on behalf of a Slack user._

### Environment Details

* SDV version: 1.2.0
* Python version: 3.10
* Operating System: Linux (Google Colab)

### Error Description
I am trying to use a PARSynthesizer on multi-sequence data that has a datetime context column (birthdate). While it is able to train, there is an error when sampling synthetic data.

### Steps to reproduce

```python
import pandas as pd
from sdv.metadata import SingleTableMetadata
from sdv.sequential import PARSynthesizer

# create 2 fake sequences for different users
# the 'birthdate' column is a context column because it does not change
data = pd.DataFrame(data={
    'user_id': ['ID_00']*5 + ['ID_01']*5,
    'birthdate': ['1995-05-06']*5 + ['1982-01-21']*5,
    'timestamp': ['2023-06-21', '2023-06-22', '2023-06-23', '2023-06-24' , '2023-06-25']*2,
    'heartrate': [67, 66, 68, 65, 64, 80, 82, 91, 88, 84]
})

metadata = SingleTableMetadata.load_from_dict({
    'columns': {
        'user_id': { 'sdtype': 'id', 'regex_format': 'ID_[0-9]{2}' },
        'birthdate': { 'sdtype': 'datetime', 'datetime_format': '%Y-%m-%d' },
        'timestamp': { 'sdtype': 'datetime' , 'datetime_format': '%Y-%m-%d' },
        'heartrate': { 'sdtype': 'numerical' }
    },
    'sequence_key': 'user_id',
    'sequence_index': 'timestamp'
})

synth = PARSynthesizer(
    metadata,
    epochs=50,
    verbose=True,
    context_columns=['birthdate'])

synth.fit(data)
synth.sample(num_sequences=2)
```

Output
```
TypeError: Cannot cast DatetimeArray to dtype float64
```

### Full Stack Trace
[stack_trace.txt](https://github.com/sdv-dev/SDV/files/11910541/stack_trace.txt)

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #1485 
CU-85ztdkk6d

Since this datetime is being read incorrectly to the context model there are were a couple approaches:
- Do not transform the column before it gets to context synthesizer (unfortunately this causes errors when trying to map the context synthesizer values in the par model)
- Update transformers in the context model (cannot easily swap the transformers before validation occurs)
- Change the type of the context metadata to match

Went with last one because the context metadata is a deep copy of the metadata and is not surfaced/used to any other functionality besides fitting the context model.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-85ztdkk6d SDV - In `PARSynthesizer` I cannot pass in datetime context (`TypeError` during sampling) #1485](https://app.clickup.com/t/85ztdkk6d)

File: sdv/sequential/par.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ def _get_context_metadata(self):
 
         for column in context_columns:
             context_columns_dict[column] = self.metadata.columns[column]
+            # Context datetime SDTypes for PAR have already been converted to float timestamp
```
Comment (frances-h) [inline sdv/sequential/par.py:76]: Should we add a check here to confirm that the datetime has actually been converted to float first before we change the sdtype?

Comment (lajohn4747) [reply sdv/sequential/par.py]: Added!

File: sdv/sequential/par.py [hunk 2]
```diff
@@ -344,6 +344,12 @@ def _fit_context_model(self, transformed):
             context[constant_column] = 0
             context_metadata.add_column(constant_column, sdtype='numerical')
 
+        for column in self.context_columns:
+            # Context datetime SDTypes for PAR have already been converted to float timestamp
+            if context_metadata.columns[column]['sdtype'] == 'datetime':
+                if pd.api.types.is_numeric_dtype(context[column]):
+                    context_metadata.update_column(column, sdtype='numerical')
```
Comment (amontanez24) [inline sdv/sequential/par.py:351]: can we add unit test coverage for this part as well?

**Review by frances-h — APPROVED** [review]

File: tests/unit/sequential/test_par.py [hunk 3]
```diff
@@ -463,6 +463,49 @@ def test__fit_context_model_with_context_columns(self, gaussian_copula_mock):
...
+        data['time'] = pd.to_datetime(data['time'])
+        data['time'] = data['time'].apply(lambda x: x.timestamp())
+        par = PARSynthesizer(metadata, context_columns=['time'])
+        initial_synthesizer = Mock()
+        context_metadata = SingleTableMetadata.load_from_dict({
+            'columns': {'time': {'sdtype': 'datetime'}, 'name': {'sdtype': 'id'}}
+        })
+        par._context_synthesizer = initial_synthesizer
+        par._get_context_metadata = Mock()
+        par._get_context_metadata.return_value = context_metadata
```
Comment (amontanez24) [inline tests/unit/sequential/test_par.py:486]: I think you can do 
```suggestion
        par._get_context_metadata = Mock(return_value=context_metadata)
```

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**